### PR TITLE
release 3.0.0

### PR DESCRIPTION
**Pull request by jmphilli**

### CODE REVIEW

**Comment by danielhochman** [comment]
bump the __init__version!

File: docs/release_notes.rst [hunk 1]
```diff
@@ -1,12 +1,12 @@
 Release Notes
 =============
 
-v2.1.6
```
Comment (danielhochman) [inline docs/release_notes.rst]: wut

File: docs/release_notes.rst [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+Now object access is possible and recommended. See [here](https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/tests/test_attributes.py#L671) for a test example.
+
+``UnicodeSetAttributes`` do not json serialize or deserialize anymore.
+If you have any ``UnicodeSetAttributes`` persisted in dynamo you will need to manage the json encoding and decoding in application if you were relying on that functionality.
+
+* Performance enhancements for the ``UTCDateTimeAttribute`` deserialize method. (#277)
+* Fix to ignore null checks for batch delete (#283)
+* Fix for ``ListAttribute`` and ``MapAttribute`` serialize (#286)
+* Fix for ``MapAttribute`` pythonic access (#292) This is a breaking change
+* Deprecated the json decode in ``;UnicodeSetAttribute`` (#294)
```
Comment (danielhochman) [inline docs/release_notes.rst]: what's with the semicolon?

File: docs/release_notes.rst [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
 =============
 
+v3.0.0
+-----
+
+:date: 2017-06-09
+
+This is a major release with breaking changes.
+
+``MapAttribute`` now allows pythonic access when recursively defined.
```
Comment (danielhochman) [inline docs/release_notes.rst]: i'm not sure how these double backticks will render in the output

File: docs/release_notes.rst (hunk 2 above)
Comment (jmphilli) [inline docs/release_notes.rst]: it's right next to backtick on dvorak -- typo

File: docs/release_notes.rst [hunk 4]
```diff
@@ -1,6 +1,39 @@
...
+
+``UnicodeSetAttributes`` do not json serialize or deserialize anymore.
+We deprecated the functionality of json serializing as of ``1.6.0`` but left the deserialization functionality in there so people could migrate away from the old functionality. 
+If you have any ``UnicodeSetAttributes`` persisted in dynamo you will need to migrate your data or manage the json encoding and decoding in application if you were relying on that functionality.
+
+* Performance enhancements for the ``UTCDateTimeAttribute`` deserialize method. (#277)
+* Fix to ignore null checks for batch delete (#283)
+* Fix for ``ListAttribute`` and ``MapAttribute`` serialize (#286)
+* Fix for ``MapAttribute`` pythonic access (#292) This is a breaking change
+* Deprecated the json decode in ``UnicodeSetAttribute`` (#294)
```
Comment (garrettheel) [inline docs/release_notes.rst]: This is potentially breaking too

Comment (jmphilli) [inline docs/release_notes.rst]: i documented that at the top

Comment (jmphilli) [inline docs/release_notes.rst]: oh add it to the list sure

File: docs/release_notes.rst [hunk 5]
```diff
@@ -1,6 +1,40 @@
...
+v3.0.0
+-----
+
+:date: 2017-06-09
+
+This is a major release with breaking changes.
+
+``MapAttribute`` now allows pythonic access when recursively defined.
+If you were not using the ``attr_name=`` kwarg then you should have no problems upgrading.
+Previously defined non subclasesd ``MapAttributes`` (raw ``MapAttributes``) that were members of a subclassed ``MapAttribute`` (typed ``MapAttributes``) would have to be accessed like a dictionary.
```
Comment (garrettheel) [inline docs/release_notes.rst]: s/subclasesd/subclassed

File: docs/release_notes.rst [hunk 6]
```diff
@@ -1,6 +1,40 @@
...
+Now object access is possible and recommended. See [here](https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/tests/test_attributes.py#L671) for a test example.
+Object notation with the ``attr_name`` also known as the DynamoDB name will now throw an ``AttributeError``.
```
Comment (garrettheel) [inline docs/release_notes.rst]: Access via the ``attr_name``, also known as the DynamoDB name, will now throw an ``AttributeError``

File: docs/release_notes.rst [hunk 7]
```diff
@@ -1,6 +1,40 @@
...
+
+``MapAttribute`` now allows pythonic access when recursively defined.
+If you were not using the ``attr_name=`` kwarg then you should have no problems upgrading.
+Previously defined non subclassed ``MapAttributes`` (raw ``MapAttributes``) that were members of a subclassed ``MapAttribute`` (typed ``MapAttributes``) would have to be accessed like a dictionary.
+Now object access is possible and recommended. See [here](https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/tests/test_attributes.py#L671) for a test example.
+Access via the ``attr_name``, also known as the DynamoDB name, will now throw an ``AttributeError``.
+
+``UnicodeSetAttributes`` do not json serialize or deserialize anymore.
+We deprecated the functionality of json serializing as of ``1.6.0`` but left the deserialization functionality in there so people could migrate away from the old functionality. 
+If you have any ``UnicodeSetAttributes`` persisted in dynamo you will need to migrate your data or manage the json encoding and decoding in application if you were relying on that functionality.
```
Comment (garrettheel) [inline docs/release_notes.rst]: technically it's if they have any ``UnicodeSetAttributes`` that have _not_ been written with a pynamodb version >= 1.6.0. I'd also say "manage json encoding/decoding with a custom attribute in application"

**Pull request merged**